### PR TITLE
Added WebAuth Auth0 Session Clear

### PR DESCRIPTION
**Pull request by cocojoe**
This call is silent/hidden with no redirect as it's not a web app so the only requirement, there is a callback so the user can do something if they want to from the result.

### CODE REVIEW

File: Auth0/SafariWebAuth.swift [hunk 1]
```diff
@@ -194,3 +201,19 @@ private func generateDefaultState() -> String? {
     guard result == 0 else { return nil }
     return data.a0_encodeBase64URLSafe()
 }
+
+private class SilentSFSafariViewController: SFSafariViewController, SFSafariViewControllerDelegate {
```
Comment (hzalaz) [inline Auth0/SafariWebAuth.swift]: Let's move this so a diff file

**Comment by cocojoe** [comment]
@hzalaz I would say it's as good as transparent, you can see it faintly at the top.
logout

File: Auth0/WebAuth.swift [hunk 2]
```diff
@@ -200,4 +200,20 @@ public protocol WebAuth: Trackable, Loggable {
...
+     ```
+     Auth0
+        .webAuth()
+        .clearSession(useFederated: false) { print($0) }
+     ```
+     
+     - parameter federated: Bool to enable signout of the IdP
+     - parameter callback: callback called with the result of the logout
+     */
+    func clearSession(useFederated federated: Bool, callback: @escaping (Bool) -> Void)
```
Comment (hzalaz) [inline Auth0/WebAuth.swift]: Just `federated` and probably make it an optional  parameter

File: Auth0/WebAuth.swift [hunk 3]
```diff
@@ -200,4 +200,20 @@ public protocol WebAuth: Trackable, Loggable {
      - parameter callback: callback called with the result of the WebAuth flow
      */
     func start(_ callback: @escaping (Result<Credentials>) -> Void)
+
+    /**
+     Clear the SSO Cookie in Auth0 and optionally signout from IdP.
```
Comment (hzalaz) [inline Auth0/WebAuth.swift]: Removes Auth0 session and optionally allows to remove the Identity Provider session.

File: Auth0/WebAuth.swift [hunk 4]
```diff
@@ -200,4 +200,20 @@ public protocol WebAuth: Trackable, Loggable {
...
+     More info: https://auth0.com/docs/logout
```
Comment (hzalaz) [inline Auth0/WebAuth.swift]: Use `note` or `seeAlso` like in the rest of the docs of the public API

File: Auth0/SilentSafariViewController.swift [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+    required init(url URL: URL, callback: @escaping (Bool) -> Void) {
+        super.init(url: URL, entersReaderIfAvailable: false)
+        self.onResult = callback
+        self.delegate = self
+        self.view.alpha = 0.05 // Apple does not allow invisible SafariViews, this is the threshold.
+        self.modalPresentationStyle = .overCurrentContext
+    }
+
+    func safariViewController(_ controller: SFSafariViewController, didCompleteInitialLoad didLoadSuccessfully: Bool) {
```
Comment (hzalaz) [inline Auth0/SilentSafariViewController.swift]: What if it fails? should we handle that case?

File: Auth0.podspec [hunk 6]
```diff
@@ -10,6 +10,7 @@ web_auth_files = [
   'Auth0/WebAuthError.swift',
   'Auth0/SafariWebAuth.swift',
   'Auth0/SafariSession.swift',
+  'SilentSafariViewController.swift',
```
Comment (hzalaz) [inline Auth0.podspec]: Why its in the root and not with the rest ?

File: Auth0/WebAuth.swift (hunk 4 above)
Comment (cocojoe) [inline Auth0/WebAuth.swift]: Annoyingly, `seeAlso` appears to be broken in Xcode, few bug reports out there.

File: Auth0/WebAuth.swift (hunk 2 above)
Comment (cocojoe) [inline Auth0/WebAuth.swift]: Originally was going to default to `false` (not a fan of optional booleans) then thought the user should understand what method they are calling.

File: Auth0/SilentSafariViewController.swift (hunk 5 above)
Comment (cocojoe) [inline Auth0/SilentSafariViewController.swift]: Not sure I understand, the controller is dismissed and the success status `didLoadSuccessfully` is returned in the callback.
Are you saying we should wrap it into an Error type for coding style?

Comment (hzalaz) [inline Auth0/SilentSafariViewController.swift]: If the url is invalid? if there is no internet connection? what will happened then?

File: Auth0/WebAuth.swift (hunk 2 above)
Comment (hzalaz) [inline Auth0/WebAuth.swift]: All good making it mandatory then

File: Auth0/WebAuth.swift (hunk 4 above)
Comment (hzalaz) [inline Auth0/WebAuth.swift]: Yeah, they changed the parser yet again

File: Auth0/SilentSafariViewController.swift (hunk 5 above)
Comment (cocojoe) [inline Auth0/SilentSafariViewController.swift]: If the URL called in the SFSafariViewController responds with anything other than HTTP 200 OK the callback will return false. (Have also manually tested this).

**Pull request merged**